conda_smithy/lint_recipe.py

```diff
@@ -147,7 +152,29 @@ def find_local_config_file(recipe_dir, filename):
     return found_filesname[0] if found_filesname else None
 
 
-def lintify_forge_yaml(recipe_dir=None) -> (list, list):
+def _forge_yaml_hint_extra_fields(forge_yaml: dict) -> List[str]:
+    """
+    Identify unexpected keys in the conda-forge.yml file.
+    This only works if extra="allow" is set in the Pydantic sub-model where the unexpected key is found.
```

Comment:
This is adding more hints than necessary. For eg: `AzureRunnerSettings` and `CondaBuildConfig` already has `extra=allow`, but we don't want hints there.

Reply:
Agreed. The hinting is now configurable in the model and disabled for the two models you mentioned.